lib/rack/proxy.rb

```diff
@@ -97,11 +98,15 @@ def perform_request(env)
         target_response.use_ssl = use_ssl
         target_response.read_timeout = read_timeout
         target_response.verify_mode = OpenSSL::SSL::VERIFY_NONE if use_ssl && ssl_verify_none
+        target_response.ssl_version = @ssl_version if @ssl_version
       else
-        start_opts = use_ssl ? {:use_ssl => use_ssl} : {}
-        start_opts[:verify_mode] = OpenSSL::SSL::VERIFY_NONE if use_ssl && ssl_verify_none
-        start_opts[:read_timeout] = read_timeout
-        target_response = Net::HTTP.start(backend.host, backend.port, start_opts) do |http|
+        serial = Net::HTTP.new(backend.host, backend.port)
```

Comment:
I would rename `serial` to `http`, and below change the line `target_response = serial.start() do |http|`
to just `target_response = serial.start do` (without the parenthesis and the block arg). The arg passed to the block is the same exact object that is assigned to `http` anyway.